apps/ledger-live-mobile/src/components/Storyly/StoryBar.tsx

```diff
@@ -0,0 +1,178 @@
+  title: "",
+  index: 0,
+  seen: true,
+  stories: [{ seen: true, id: undefined }],
+};
+
+const PlaceholderContent: StoryGroupInfo[] = new Array(5).fill(
+  PlaceholderStoryGroup,
+);
+
+type StoryInfo = { seen: boolean; id?: string };
+
+const ScrollView = styled.ScrollView.attrs({ horizontal: true })`
+  flex: 1;
+`;
+
+const AnimatedStoryGroupWrapper = Animated.createAnimatedComponent(
+  styled(Flex).attrs((p: { isLast: boolean }) => ({
+    mr: p.isLast ? 0 : "14px",
+  }))``,
+);
+
+const StoryBar: React.FC<Props> = props => {
+  const { keepOriginalOrder = false, style } = props;
+
+  const storylyRef = useRef<Storyly>(null);
+  const [storyGroupList, setStoryGroupList] = useState<StoryGroupInfo[]>(
+    PlaceholderContent,
+  );
+
+  const [refreshingStorylyState, setRefreshingStorylyState] = useState<boolean>(
+    false,
+  );
+
+  const handleFail = useCallback(() => {
+    setStoryGroupList(PlaceholderContent);
+  }, [setStoryGroupList]);
+
+  const handleLoad = useCallback(
+    (event: Storyly.StoryLoadEvent) => {
+      setRefreshingStorylyState(false);
+      const newStoryGroupList = event.storyGroupList.map(group => ({
+        ...group,
+        stories: group.stories.map(s => ({ id: s.id, seen: s.seen })),
+      }));
+      if (!isEqual(storyGroupList, newStoryGroupList))
+        setStoryGroupList(newStoryGroupList);
+    },
+    [storyGroupList, setStoryGroupList, setRefreshingStorylyState],
+  );
+
+  const handleStoryGroupPressed = useCallback(
+    (storyGroupId?: string, storyId?: string) => {
+      if (!storyGroupId || !storyId) return;
+      storylyRef?.current?.openStoryWithId(
+        // @ts-ignore the typing of storyly is broken, the iOS native module
+        // actually expects numbers or it will crash.
+        Platform.OS === "ios"
+          ? Number.parseInt(storyGroupId, 10)
+          : storyGroupId,
+        Platform.OS === "ios" ? Number.parseInt(storyId, 10) : storyId,
+      );
+    },
+    [storylyRef],
+  );
+
+  const handleEvent = useCallback(
+    (event: Storyly.StoryEvent) => {
+      if (["StoryGroupClosed", "StoryGroupCompleted"].includes(event.event)) {
+        /** These are all the events that can be triggered when the full screen
+         * story view exits.
+         * The only way to get fresh data about the state of the stories is to
+         * mount a new Storyly component that will trigger a call of onLoad with
+         * up to date data (which story groups is "seen", their order etc.)
+         */
+        setRefreshingStorylyState(true);
+      }
+    },
+    [setRefreshingStorylyState],
+  );
+
+  const renderedStoryGroups = useMemo(
+    () =>
+      storyGroupList
+        .sort((a, b) => (keepOriginalOrder ? a.index - b.index : 1)) // storyly reorders the array by default
+        .map((storyGroup, index, arr) => {
+          const firstUnseenStoryId =
+            storyGroup.stories?.find(story => !story.seen)?.id ??
+            storyGroup.stories[0]?.id;
```

Comment:
the `storyGroup.stories[0]` is the first unseen story only if let storyly reorder the array, no ? 

If we force our order (by index), the index 0 might not be the first unseen ?

Or am i wrong ?

Reply:
I guess it's just bad naming, here I'm already looking for the first unseen one (with `.find()`) and then if there is none (`??`) resorting to taking the first one in the array.
The variable should be called `nextStoryToShowID` or something like that I guess.

Reply:
Also actually the stories themselves are not reordered, it's the story groups that can be reordered.

Reply:
Ah yes, ok ! Thanks - yeah maybe `nextStoryToShowId` is better